### PR TITLE
Pass service opts to make-service-method

### LINKED ISSUE #176: Creating uberwars with ring-1.6.0 async support
I’m trying to deploy an app to my jetty9 server using the async support in ring-1.6.0-beta5. While the app works fine with `lein ring server-headless`, the version created with `lein ring uberwar` doesn’t seem to work:

```
clojure.lang.ArityException: Wrong number of args (1) passed to: core/handler
```

My code and the stack trace are at https://gist.github.com/sander/73b37a2684dfdbb0c3d1106c10525253. I am using:
- OS X 10.11.6 (15G31)
- JRE 1.8.0_101-b13
- Jetty 9.3.11.v20160721
- lein-ring 0.9.7
- ring 1.6.0-beta5

The error seems to be expected, since README.md tells that the `:adapter` option has no effect when deploying the app as a war file. I use `:adapter {:async? true}` to make the embedded server run the handler asynchronously.

Is there another way to create a war file that will be handled asynchronously by jetty9?

### PR DESCRIPTION
**Pull request by MichaelBlume**
Fix #176

### CODE REVIEW

**Comment by weavejester** [comment]
Thanks for the patch. Could you change the commit message so that it's capitalized?

Also I think that we probably want to use a `:service` option rather than co-opt the `:adapter`, as there isn't necessarily any overlap between the two.

**Comment by MichaelBlume** [comment]
Better?

(Should probably be noted that right now, this won't work unless the user
explicitly depends on the beta version of ring-servlet)

**Comment by weavejester** [comment]
That'll work, but you added an extra period to the end of the commit message. Can you remove that? So something like:

    Pass service options to make-service-method

    Fixes #176.

I follow the usual [seven rules](http://chris.beams.io/posts/git-commit/#seven-rules) when writing commit messages. It helps to keep the project history consistent.

**Comment by MichaelBlume** [comment]
Done =)

**Comment by weavejester** [comment]
Thanks! Could you possibly add a small description of the `:service` key to to the "General Options" section of the README? Maybe something like:

> * `:service` - A map of options to be passed to the `make-service-method` function. This only has an effect when you're deploying your application as a war-file.

Could you also change the commit message to the text I quoted? i.e. "fixes" instead of "fix" since it's describing what the previous line does.

**Comment by MichaelBlume** [comment]
Done =)

**Comment by weavejester** [comment]
You don't seem to have committed the changes to the README?

**Comment by MichaelBlume** [comment]
Ah, sorry, somehow I only saw the second half of your reply.

I'm actually not sure I'm happy with this. An async ring handler will need to be treated as an async handler whether it's started with lein ring service or with installation as a jetty servlet. Shouldn't the user only need to specify this once?

**Comment by weavejester** [comment]
We could make `:async?` a top-level option, the same as `:port`.

**Comment by MichaelBlume** [comment]
Yeah, makes sense.

**Comment by weavejester** [comment]
Now that I think about it, that might be the best option. If you want to go ahead and make that change, I think I'd be happy to accept it.

**Comment by MichaelBlume** [comment]
Given that there's only one service option -- async? -- and we want it to be set at top level, I think I'm going to leave off allowing the user to pass in a map of service options; we may want it later, but there's no good reason for it now.

**Comment by MichaelBlume** [comment]
Top level async option ready for review

File: src/leiningen/ring/war.clj [hunk 1]
```diff
@@ -146,6 +146,7 @@
   (let [init-sym    (get-in project [:ring :init])
         destroy-sym (get-in project [:ring :destroy])
         handler-sym (get-in project [:ring :handler])
+        service     (select-keys (:ring project) [:async?])
```
Comment (weavejester) [inline src/leiningen/ring/war.clj]: Maybe just change this to `async?`, and have the following clause as `(if async? {:async? async?})`

**Comment by MichaelBlume** [comment]
Done

**Comment by weavejester** [comment]
This should be perfect now, I just need to find some time to test it. :)

**Comment by erdos** [comment]
Hi, will you get a chance to look at this PR? I would love to use it in production environment when possible. Thanks

**Comment by MichaelBlume** [comment]
This looks pretty definitely safe to me, I'm gonna merge and release if you have no objections.

**Comment by weavejester** [comment]
Go ahead :)

**Comment by MichaelBlume** [comment]
Thanks =)

@erdos this is now available in lein-ring 0.11.0

**Comment by erdos** [comment]
Thank you!

**Pull request merged**